prettier-el.js

```diff
@@ -174,54 +183,83 @@ function getGlobalPrettier() {
+ * directories.
  *
- * @param {!string} directory
- * @return {!PrettierAPI|null}
+ * @param {!string} directory  The directory where to start looking.
+ * @param {!Array<string>} fileNames  All file names to try in each directory.
+ * @return {!string|null}  Path to the file found, or null if not found.
  */
-function getLocalPrettier(directory) {
-  const prettierCandidate = path["join"](directory, "node_modules", "prettier");
+function findFileInAncestry(directory, fileNames) {
+  for (let i = 0; i < fileNames.length; ++i) {
+    const candidate = path["join"](directory, fileNames[i]);
+    if (fs["existsSync"](candidate)) {
+      return candidate;
+    }
+  }
+
+  const parent = path["dirname"](directory);
+  if (parent !== directory) {
+    return findFileInAncestry(parent, fileNames);
+  }
+
+  return null;
+}
+
+/**
+ * Try requiring the Prettier package using the given require function.
+ *
+ * @param {!Function} targetRequire  The require function to use.
+ * @return {PrettierAPI}  The Prettier package if found, or null if not found.
+ */
+function tryRequirePrettier(targetRequire) {
   try {
-    const stat = fs["statSync"](prettierCandidate);
-    return stat.isDirectory() ? externalRequire(prettierCandidate) : null;
+    return targetRequire("prettier");
   } catch (e) {
-    if (e.code === "ENOENT") return null;
+    if (
+      Array.prototype.includes.call(
+        ["MODULE_NOT_FOUND", "UNDECLARED_DEPENDENCY"],
+        e.code
+      )
+    ) {
+      return null;
+    }
     throw e;
   }
 }
 
 /**
- * Return the Prettier package under Yarn Plug'n'Play if either `.pnp.js` or
- * `.pnp.cjs` exists. Inject PnP API into the Node.js environment only once.
- * This side effect is unlikely to be a problem at present but we recommend to
- * use project-specific Node.js executable with any tool such as `direnv`.
+ * Find locally installed Prettier, or null if not found.
  *
- * @param {!string} directory
- * @return {!PrettierAPI|null}
+ * @param {!string} directory  The directory for which to find a local Prettier installation.
+ * @return {PrettierAPI}  The Prettier package if found, or null if not found.
  */
-function getYarnPnpifyedLocalPrettier(directory) {
-  const yarnPnpCandidate = path["join"](directory, ".pnp");
-  if (
-    fs["existsSync"](yarnPnpCandidate + ".js") ||
-    fs["existsSync"](yarnPnpCandidate + ".cjs")
-  ) {
-    try {
-      const m = externalRequire("module");
-      const createRequire = m["createRequire"] || m["createRequireFromPath"];
-      if (!createRequire) {
-        throw new Error("You should upgrage Node.js v10.12.0 or above.");
-      }
-      if (!process["versions"]["pnp"]) {
-        externalRequire(yarnPnpCandidate)["setup"]();
-      }
-      const targetRequire = createRequire(yarnPnpCandidate);
-      return targetRequire(targetRequire["resolve"]("prettier"));
-    } catch (e) {
-      if (e.code === "ENOENT") return null;
-      throw e;
-    }
+function getLocalPrettier(directory) {  
```

Comment:
There are trailing white spaces.

Reply:
Do you mean the double space before `The`? I did that on purpose, to separate the description from the argument name. Or what do you mean?

Reply:
I'm sorry for not explaining enough.
These spaces before `{`
```
function getLocalPrettier(directory) {<space><space>
```

Reply:
Oh, yes you're right. That's really weird. I'm using `whitespace-cleanup-mode` and I could have sworn that GitHub highlights trailing spaces in the diff view. So it seems that two completely unrelated tools let me down!

Fixing this now...